runtime-testsuite/resources/org/antlr/v4/test/runtime/descriptors/ParserExec/ListLabelsOnRuleRefStartOfAlt.txt

```diff
@@ -0,0 +1,24 @@
+[notes]
+Checks that this compiles; see https://github.com/antlr/antlr4/issues/2016
+
+[type]
+Parser
+
+[grammar]
+grammar Test;
+expression: op=NOT args+=expression
+          | args+=expression (op=AND args+=expression)+
+          | args+=expression (op=OR args+=expression)+
+          | IDENTIFIER
+          ;
+AND : 'and' ;
+OR : 'or' ;
+NOT : 'not' ;
+IDENTIFIER : [a-zA-Z_][a-zA-Z0-9_]* ;
+WS : [ \t\r\n]+ -> skip ;
+
+[start]
+expression
+
+[input]
+a and b
```

Comment:
I don't think it's correct test for checking of element labels. It checks if input is being parsed but it doesn't check the final value of `args` label.

Reply:
Agreed. The actual error however was failure to compile. Let me add some output generation.

Reply:
wow. We literally do not use the writeList testing template anywhere... ugh. I will do something expedient to validate that we are getting a list. I think there is a template for that.